k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapFolder.java

```diff
@@ -167,6 +167,32 @@ public void open(int mode) throws MessagingException {
         }
     }
 
+    public long getCurrentUidValidity() throws MessagingException {
```

Comment:
can't this be done in an easier way by simply calling `open` if the folder is not yet opened? this already open the folder, and fetches a `SelectOrExamineResponse`, which can be used to parse and store the uidvalidity, so in cases the folder is already opened we don't have to do any server communication at all